rotkehlchen/tests/utils/history_base_entry.py

```diff
@@ -16,39 +20,104 @@
 from rotkehlchen.chain.evm.types import string_to_evm_address
 from rotkehlchen.constants import ONE
 from rotkehlchen.constants.assets import A_DAI, A_ETH, A_ETH2, A_USDT
-from rotkehlchen.db.history_events import DBHistoryEvents
 from rotkehlchen.fval import FVal
-from rotkehlchen.tests.utils.api import api_url_for, assert_proper_response_with_result
+from rotkehlchen.tests.utils.api import (
+    api_url_for,
+    assert_proper_response_with_result,
+    assert_simple_ok_response,
+)
 from rotkehlchen.types import Location, TimestampMS, deserialize_evm_tx_hash
 
 if TYPE_CHECKING:
     from rotkehlchen.api.server import APIServer
 
 
+def assert_editing_works(
+        entry: HistoryBaseEntry,
+        rotkehlchen_api_server: 'APIServer',
+        sequence_index: int,
+):
+    entry.sequence_index = sequence_index
+    entry.timestamp = TimestampMS(1639924575000)
+    entry.event_type = HistoryEventType.DEPOSIT
+    entry.asset = A_USDT
+    entry.balance = Balance(amount=FVal('1500.1'), usd_value=FVal('1499.45'))
+    entry.location_label = '0x9531C059098e3d194fF87FebB587aB07B30B1306'
+    entry.notes = 'Deposit stuff for staking somewhere'
+    entry.event_subtype = HistoryEventSubType.NONE
+    json_data = entry_to_input_dict(entry, include_identifier=True)
+    response = requests.patch(
+        api_url_for(rotkehlchen_api_server, 'historyeventresource'),
+        json=json_data,
+    )
+    assert_simple_ok_response(response)
+    assert entry.identifier is not None
+
+
+def pop_useless_fields(serialized_event: dict[str, Any], fields: tuple[str, ...]):
+    for field in fields:
+        serialized_event.pop(field)
+
+
 def entry_to_input_dict(
-        entry: 'EvmEvent',
+        entry: 'HistoryBaseEntry',
         include_identifier: bool,
 ) -> dict[str, Any]:
     """
-    Converts an EvmEvent entry into a dictionary, optionally including the event identifier.
+    Converts a HistoryBaseEntry into a dictionary, optionally including the event identifier.
     """
     serialized = entry.serialize()
     if include_identifier:
         assert entry.identifier is not None
         serialized['identifier'] = entry.identifier
     else:
         serialized.pop('identifier')  # there is `identifier`: `None` which we have to remove
-    serialized.pop('event_identifier')
-    serialized.pop('entry_type')
+
+    if entry.entry_type == HistoryBaseEntryType.EVM_EVENT:
+        serialized.pop('event_identifier')
+    elif entry.entry_type == HistoryBaseEntryType.ETH_WITHDRAWAL_EVENT:
+        pop_useless_fields(serialized, (
+            'event_identifier',
+            'sequence_index',
+            'location',
+            'asset',
+            'event_type',
+            'event_subtype',
+            'notes',
+        ))
```

Comment:
since you have to name the fields that need to be removed... why not create a mapping with the fields that you want to mantain when removing multiples? If you end up adding more fields we will need to update this